src/main/resources/db/migration/V3_00__create_application_tables.sql

```diff
@@ -0,0 +1,73 @@
+CREATE TABLE tmp_visit (
+            id serial4 NOT NULL,
+            prison_id int4 NOT NULL,
+            prisoner_id varchar(80) NOT NULL,
+            session_slot_id int4 NULL,
+            reference text NOT NULL,
+            application_reference varchar(40) NULL,
+            visit_type varchar(80) NOT NULL,
+            visit_room varchar(255) NOT NULL,
+            visit_status varchar(80) NOT NULL,
+            visit_restriction varchar(80) NOT NULL,
+            outcome_status varchar(40) NULL,
+            created_by varchar(60) NOT NULL DEFAULT 'NOT_KNOWN'::character varying,
+            updated_by varchar(60) NULL,
+            cancelled_by varchar(60) NULL,
+            create_timestamp timestamp NULL DEFAULT CURRENT_TIMESTAMP,
+            modify_timestamp timestamp NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+-- Create new application table, this is done because updating the existing tmp table will take much long than creating a new table
+CREATE TABLE application (
+            id int4 NOT NULL,
+            prison_id int4 NULL,
+            prisoner_id varchar(80) NULL,
+            session_slot_id int4 NULL,
+            reference text UNIQUE NULL,
+            booking_reference text NULL,
+            visit_type varchar(80) NULL,
+            restriction varchar(80) NULL,
+            created_by varchar(60) NULL,
+            create_timestamp timestamp NULL,
+            modify_timestamp timestamp NULL
+);
+
+-- Create session slot table
+CREATE TABLE session_slot
```

Comment:
will this be created for all dates when we create a session template?

Reply:
Narr this will be created in a lazy fashion when the first booking is booked for that slot,... after that it will just reuse for that slot.